src/asm/generate_commands_dis.c

```diff
@@ -1,22 +1,117 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   generate_commands_dis.c                            :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: jleann <marvin@42.fr>                      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2019/11/02 02:28:44 by jleann            #+#    #+#             */
+/*   Updated: 2019/11/02 02:28:44 by jleann           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
 #include "disassembler.h"
 
-static void	read_command(t_command *cmd, char *exec_code, size_t *cur)
+inline static uint8_t	get_par_type(uint8_t ops, uint8_t id)
+{
+	return ((uint8_t)(ops >> (uint8_t)(6 - id * 2)) & 3U);
+}
+
+static void				read_params(t_command *cmd, void *exc, size_t *cr,
+									size_t cap)
+{
+	uint8_t	cur_par;
+	t_op	*op;
+
+	op = get_op_info(cmd->cmd_type);
+	cur_par = 0;
+	while (cur_par < op->params_num)
+	{
+		if (op->paramtypes[cur_par] == T_REG)
+		{
+			cmd->params[cur_par].data = read_int8_mem(exc, cr, cap);
+			cmd->params[cur_par].type = T_REG;
+		}
+		else if (op->paramtypes[cur_par] == T_IND)
+		{
+			cmd->params[cur_par].data = read_int8_mem(exc, cr, cap);
+			cmd->params[cur_par].type = T_IND;
```

Comment:
Feels like redundant code. may be shorten for two ifs and one initiation of cmd->params[].type

Reply:
Not in this case - each if corresponds to each variable type, seems correct for me.